skiko/src/macosMain/kotlin/org/jetbrains/skiko/SkiaLayer.macos.kt

```diff
@@ -105,13 +111,60 @@ actual open class SkiaLayer() {
             }
             override fun keyDown(event: NSEvent) {
                 skikoView?.onKeyboardEvent(toSkikoEvent(event, SkikoKeyboardEventKind.DOWN))
+                keyEvent = event
+                interpretKeyEvents(listOf(event))
             }
             override fun flagsChanged(event: NSEvent) {
                 skikoView?.onKeyboardEvent(toSkikoEvent(event))
             }
             override fun keyUp(event: NSEvent) {
                 skikoView?.onKeyboardEvent(toSkikoEvent(event, SkikoKeyboardEventKind.UP))
             }
+            
+            private var keyEvent: NSEvent? = null
+            private var markedText: String = ""
+            private val kEmptyRange = NSMakeRange(NSNotFound.toULong(), 0.toULong())
```

Comment:
`NSNotFound.convert()` may be better here and in other places.

Reply:
Done.